src/Package/Impl/DataInspect/VisualGrid/VisualGridScroller.cs

```diff
@@ -106,18 +106,18 @@ internal sealed class VisualGridScroller {
                         // upperbound prevents infinite loop in case scroll commands is queued fast and endlessly, which happens only in theory
                         continue;
                     } else {
-                        for (int i = 0; i < batch.Count; i++) {
+                        for (var i = 0; i < batch.Count; i++) {
                             if (cancellationToken.IsCancellationRequested) {
                                 break;
                             }
 
-                            bool execute = true;
+                            var execute = true;
                             // if next command is same the current one, skip to next (new one) for optimization
                             if (i < (batch.Count - 1)) {
                                 if (IsRepeating(batch, i, RepeatSkip)) {
                                     execute = false;
                                 } else if (IsRepeating(batch, i, RepeatAccum)) {
-                                    batch[i + 1].Param = (double)batch[i + 1].Param + (double)batch[i].Param;
+                                    batch[i + 1].Param = Convert.ToDouble(batch[i + 1].Param) + Convert.ToDouble(batch[i].Param);
```

Comment:
`double.ToDouble` is implemented like this:
```cs
    double IConvertible.ToDouble(IFormatProvider provider)
    {
      return this;
    }
```

On the other hand, if type doesn't match, it means that it is unsynchronized with code that post commands, so it should fail on casting.

Reply:
Hmmm, I've seen object coming as `int` which caused `(double)o` cast to throw.

Reply:
This can be the case. `int` usually means row or column index. `double` is usually a pixel. So if it happens, we may send, for instance, offset as row or column index instead of offset in pixels.

Reply:
The whole system of untyped command arguments is a bug prone, but I had no time to change it into proper producer/consumer.

Reply:
Mouse wheel scroll was int